.github/workflows/build_and_test.yml

```diff
@@ -724,8 +725,9 @@ jobs:
         export MAVEN_OPTS="-Xss64m -Xmx2g -XX:ReservedCodeCacheSize=1g -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN"
         export MAVEN_CLI_OPTS="--no-transfer-progress"
         export JAVA_VERSION=${{ matrix.java }}
+        export SKIP_ENFORCER=$( [ "$JAVA_VERSION" = "8" ] && echo true || echo false )
         # It uses Maven's 'install' intentionally, see https://github.com/apache/spark/pull/26414.
-        ./build/mvn $MAVEN_CLI_OPTS -DskipTests -Pyarn -Pmesos -Pkubernetes -Pvolcano -Phive -Phive-thriftserver -Phadoop-cloud -Djava.version=${JAVA_VERSION/-ea} install
+        ./build/mvn $MAVEN_CLI_OPTS -DskipTests -Pyarn -Pmesos -Pkubernetes -Pvolcano -Phive -Phive-thriftserver -Phadoop-cloud -Djava.version=${JAVA_VERSION/-ea} -Denforcer.skip=${SKIP_ENFORCER} install
```

Comment:
`-Denforcer.skip=true` when using Java 8?

Reply:
I thought you asked for a build on Java 8 to see what happens. Skipping enforcer rule for Java 8 to check whether it causes runtime exceptions when loading threeten jar claseses.

Reply:
No, what I mean is we should do this check when user submitting code, especially for Java 8, otherwise the addition of this plugin will not have its intended effect, so I think `enforcer.skip` should always be false

Reply:
Are you looking for this? log: https://github.com/bowenliang123/spark/actions/runs/5239841838/jobs/9460075811#step:11:69

It's checked along with `mvn validate` in CI builds.

Reply:
Do you mean `./dev/test-dependencies.sh` already covered this inspection? That's great, sorry for misleading, then I think we don't need to add any extra Maven builds anymore.

Reply:
Yes, it does as you expected. Thanks.

Reply:
Thanks for your explain

Reply:
Let's revert this change :)

Reply:
Do you think I should unset Java 8  from the matrix, and skip maven builds on it?

Reply:
OK, sure will do it. And I'm curious to see a full build run on Java 8. And revert the change after that.

Reply:
Success builds with Maven on Java 8 with enforcer skipped.
https://github.com/bowenliang123/spark/actions/runs/5239841838/jobs/9460063427#step:7:7